manifests/init.pp

```diff
@@ -50,9 +50,15 @@
 
   if has_key($java::params::java, $distribution) {
     $default_package_name     = $java::params::java[$distribution]['package']
-    $default_alternative      = $java::params::java[$distribution]['alternative']
-    $default_alternative_path = $java::params::java[$distribution]['alternative_path']
-    $java_home                = $java::params::java[$distribution]['java_home']
+    if has_key($java::params::java[$distribution], 'alternative') {
+      $default_alternative = $java::params::java[$distribution]['alternative']
```

Comment:
And if not `has_key` then ... `$default_alternative` will be undefined, which will fail strict variables :)

Could you add else `$default_alternative = undef` ?